tools/mason/MasonInit.chpl

```diff
@@ -72,16 +79,65 @@ proc masonInit(args) throws {
     }
 
     if dirName == '' {
-      const cwd = getEnv("PWD");
-      var name = basename(cwd);
-      const path = '.';
-      if packageName.size > 0 then name = packageName;
-      var resName = validatePackageNameChecks(path, name);
-      name = resName;
-      validateMasonFile(path, name, show);
-      var isInitialized = validateInit(path, name, true, show);
-      if isInitialized > 0 then
-      writeln("Initialized new library project: " + basename(cwd));
+      if defaultBehavior {
+        const cwd = getEnv("PWD");
+        var name = basename(cwd);
+        const path = '.';
+        if packageName.size > 0 then name = packageName;
+        var resName = validatePackageNameChecks(path, name);
+        name = resName;
+        validateMasonFile(path, name, show);
+        var isInitialized = validateInit(path, name, true, show);
+        if isInitialized > 0 then
+        writeln("Initialized new library project: " + basename(cwd));
+      } else {
+        // check if Mason.toml file and src/moduleFile is present
+        var isMasonTomlPresent = false;
+        var isSrcPresent = false;
+        var moduleName: string;
+        if isFile('./Mason.toml') then isMasonTomlPresent = true;
+        if isDir('./src') then isSrcPresent = true;
+        // parse values from TOML File && module file
+        var defaultPackageName: string;
+        var defaultVersion: string;
+        var defaultChplVersion: string;
+        if isMasonTomlPresent {
+          const toParse = open("./Mason.toml", iomode.r);
+          const tomlFile = owned.create(parseToml(toParse));
+          if tomlFile.pathExists("brick.name") then
+            defaultPackageName = tomlFile["brick"]!["name"]!.s;
+          if tomlFile.pathExists("brick.version") then
+            defaultVersion = tomlFile["brick"]!["version"]!.s;
+          if tomlFile.pathExists("brick.chplVersion") then
+            defaultChplVersion = tomlFile["brick"]!["chplVersion"]!.s;
+        }
+        if isSrcPresent {
+          var file: string = listdir("./src");
+          moduleName = file;
+        }
+       // begin interactive session and get values input by user
+        var newPackageName, newVersion, newChplVersion: string;
+        var result = beginInteractiveSession(defaultPackageName, defaultVersion, defaultChplVersion);
+        newPackageName = result[0];
```

Comment:
It's generally better to declare your variables close to or in the same place as where you initialize them. Here, we can do it in the same place. Also, these can be const since we don't expect them to change at runtime (this applies to many variables throughout)

I would suggest either:

```chpl
const (newPackageName, newVersion, newChplVersion) = beginInteractiveSession(defaultPackageName, defaultVersion, defaultChplVersion);
```

or

```chpl
const result = beginInteractiveSession(defaultPackageName, defaultVersion, defaultChplVersion);
const newPackageName = result[0],
      newVersion = result[1],
      newChplVersion = result[2];
```